### PR TITLE
Adding ordering and tests.

### PR DESCRIPTION
**Pull request by ptpham**

### CODE REVIEW

File: ordering.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  this.rebind();
+};
+
+Ordering.prototype.clip = function(first, second) {
+  var left = this.resolve(first), right = this.resolve(second);
+  var result = this.layers.splice(left.index, right.index - left.index);
+  this.rebind();
+  return result;
+};
+
```
Comment (ptpham) [inline ordering.js]: Add more correct version of clip.

File: ordering.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+Ordering.prototype.merge = function(first, second) {
+  var target = this.resolve(first), other = this.resolve(second),
+      mapping = this.mapping, layers = this.layers;
+  layers.splice(layers.indexOf(other), 1);
+  _.each(other, function(elem) { mapping.set(elem, target); });
+  Array.prototype.push.apply(target, other);
+  this.rebind();
+};
+
+Ordering.prototype.remove = function(elem) {
```
Comment (ptpham) [inline ordering.js]: Rename.

File: ordering.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+};
+
+Ordering.prototype.clip = function(first, second) {
+  var left = this.resolve(first), right = this.resolve(second);
+  var result = this.layers.splice(left.index, right.index - left.index);
+  this.rebind();
+  return result;
+};
+
+Ordering.prototype.insert = function(elem, clip) {
```
Comment (ptpham) [inline ordering.js]: Rename.

File: ordering.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  }
+  return this.mapping.get(elem);
+};
+
+Ordering.prototype.resolveAfter = function(elem) {
+  return this.layers[this.resolve(elem).index + 1];
+};
+
+Ordering.prototype.rebind = function() {
+  _.each(this.layers, function(layer, index) { layer.index = index; });
```
Comment (ptpham) [inline ordering.js]: Compact before rebinding.

File: ordering.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+};
+
+Ordering.prototype.below = function(first, second) {
```
Comment (ptpham) [inline ordering.js]: Change to compare and return integer.

**Comment by cixzhang** [comment]
Looks good to me! :smile:

**Pull request merged**